docs/building_chains.rst

```diff
@@ -42,3 +42,41 @@ Then to initialize, you can start it up with an in-memory database:
 
   # initialize a fresh chain
   chain = chain_class.from_genesis_header(chaindb, MAINNET_GENESIS_HEADER)
+
+
+Using the LightChain object
+---------------------------
+
+The :class:`~evm.p2p.lightchain.LightChain` is like a Chain but it will also
+connect to remote peers and fetch new :class:`~evm.rlp.headers.BlockHeader` s
+as they are announced on the network. As such, it must first be configured
+with a vm_configuration, but it also requires a network_id and privkey:
+
+::
+
+  from evm.chains.mainnet import MAINNET_VM_CONFIGURATION, MAINNET_NETWORK_ID
+  from evm.p2p import ecies
+  from evm.p2p.lightchain import LightChain
+
+  DemoLightChain = LightChain.configure(
+      name='Demo LightChain',
+      privkey=ecies.generate_privkey(),
+      vm_configuration=MAINNET_VM_CONFIGURATION,
+      network_id=MAINNET_NETWORK_ID,
+  )
+
+
+And in order for it to connect to other peers in the network and fetch new
+headers, you should tell asyncio to execute its run() method:
+
+::
+
+  import asyncio
+  from evm.db.backends.memory import MemoryDB
+  from evm.db.chain import BaseChainDB
+  from evm.chains.mainnet import MAINNET_GENESIS_HEADER
+
+  chain = DemoLightChain.from_genesis_header(BaseChainDB(MemoryDB()), MAINNET_GENESIS_HEADER)
```

Comment:
The amount of imports here make me wonder how we can establish some built-in `Chain` subclasses which have suitable defaults as well as potentially environment based configuration for the `ChainDB` class and the `BaseDB` class.  This could reduce this down to 

```
from evm.chains.demo import DemoMainnetLightChain

chain = DemoMainnetLightChain()
chain.run()   # can we put the `asyncio` loop setup into this or another method?
```

Reply:
Yeah, the amount of imports is annoying, and maybe we could have a helper for the asyncio loop, but it should not be in chain.run() because different services will use different asyncio APIs to run the chain (e.g. the interactive shell uses `run_until_complete()` whereas the RPC server uses `ensure_future()`)